src/Controller/Hello.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class Hello
+{
+    use ControllerTrait;
+
+    /**
+     * @Route("/", name="homepage")
+     */
+    public function __invoke(Request $request): Response
```

Comment:
Well, a (controller) class is a (callback) callable "only" if this one implements the `__invoke()` method otherwise it's not.

But what I'm trying to say here is that `@route()` annotation is setting up the `_controller` as `App\Controller\Hello::__invoke` directly, so using `__invoke()` method is useless here (i.e. does not make any magic calls).